tools/bamclipper/bamclipper.xml

```diff
@@ -0,0 +1,86 @@
+<tool id="bamclipper" name="BAMClipper" version="@VERSION@+galaxy0">
+    <description> Remove gene-specific primer sequences from BAM alignments of PCR amplicons by soft-clipping with BEDPE file</description>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <expand macro="requirements" />
+    <command detect_errors="exit_code"><![CDATA[
+
+#set bamname = str($input1.name)
```

Comment:
Can you change your set line with one below to ensure we do not get weird file name characters.
set identifier = re.sub('[^\s\w\-]', '_', str($input1.name))

Reply:
Done!